lib/bamboo/sent_email.ex

```diff
@@ -89,7 +89,7 @@ defmodule Bamboo.SentEmail do
 
   defp do_get(id) do
     Enum.find all, nil, fn(email) ->
-      get_id(email) == id
+      get_id(email) == String.downcase(id)
```

Comment:
I'm not sure this needs to be downcast, since the URL is generated using the email id, and it should already be downcased at that point